# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/android/SessionAccessibility.cpp
# Commit: 9e82c8317cf6
# Full Hash: 9e82c8317cf60b1372add3d66e9fc735493a8ca4
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-07-22 03:25:11
# Description:
#   Bug 1778585 - Don't register non-top-level accessibles if there is no top-level regeistered. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152328
# ==============================================================================

diff -r 96609971f3cf -r 9e82c8317cf6 accessible/android/SessionAccessibility.cpp
--- a/accessible/android/SessionAccessibility.cpp	Thu Jul 21 20:03:32 2022 +0300
+++ b/accessible/android/SessionAccessibility.cpp	Thu Jul 21 16:57:04 2022 +0000
@@ -1015,15 +1015,19 @@
 
   int32_t virtualViewID = kNoID;
   if (!isTopLevel) {
+    if (sessionAcc->mIDToAccessibleMap.IsEmpty()) {
+      // We expect there to already be at least one accessible
+      // registered (the top-level one). If it isn't we are
+      // probably in a shutdown process where it was already
+      // unregistered. So we don't register this accessible.
+      return;
+    }
     // Don't use the special "unset" value (0).
     while ((virtualViewID = sIDSet.GetID()) == kUnsetID) {
     }
   }
   AccessibleWrap::SetVirtualViewID(aAccessible, virtualViewID);
 
-  MOZ_ASSERT(
-      !sessionAcc->mIDToAccessibleMap.IsEmpty() || virtualViewID == kNoID,
-      "root (kNoID) accessible should be the first one added");
   MOZ_ASSERT(!sessionAcc->mIDToAccessibleMap.Contains(virtualViewID),
              "ID already registered");
   sessionAcc->mIDToAccessibleMap.InsertOrUpdate(virtualViewID, aAccessible);
